yazi-adapter/src/emulator.rs

```diff
@@ -70,7 +72,7 @@ impl Emulator {
 		];
 		match vars.into_iter().find(|v| env_exists(v.0)) {
 			Some(var) => return var.1,
-			None => warn!("[Adapter] No special environment variables detected"),
+			None => warnings.push("[Adapter] No special environment variables detected"),
```

Comment:
It's tricky to implement it this way, which means that if any of the subsequent conditions are met and an early return occurs, this log will never be printed.